Challenge

DeVuDeveloper opened the pull request:
**Refactoring and Setup:**

- Added essential gems for enhanced functionality: `simple_form`, `turbo-rails`, `stimulus-rails`, `redis`, and `sidekiq`.
- Significantly improved page rendering by leveraging turbo frames and turbo streams.
- Implemented form validation to provide a smoother user experience.
- Conducted comprehensive tests prior to actual code implementation.
- Refactored the codebase and introduced view components for better maintainability and readability.
- Added tests before starting the problem-solving process, ensuring that new features and changes are thoroughly validated.

### **Issue 1: "Warning: Each child in an array or iterator should have a unique 'key' prop."**

**Description:** The React application is showing a warning "Each child in an array or iterator should have a unique 'key' prop." This warning occurs when rendering a list of elements, and a unique `key` attribute is not provided for each element.

**Solution:**

The issue was resolved by updating the `OrderTable` component to provide a unique `key` attribute for each `OrderRow` component that is rendered within the `.map()` function. The `key` attribute is set to the `id` of each order to ensure uniqueness.

```jsx
import React from 'react';
import OrderRow from './OrderRow';

const OrderTable = ({ orders }) => (
  <table className="table orders-table">
    <thead>
      <tr>
        <th>Order #</th>
        <th>Ordered at</th>
        <th>Pick up at</th>
        <th>Customer Name</th>
        <th>Item</th>
        <th>Qty</th>
        <th>Status</th>
        <th>Actions</th>
      </tr>
    </thead>
    <tbody>
      {orders.map((order) => (
        <OrderRow key={order.id} order={order} /> // Added a unique 'key' attribute with the id
      ))}
    </tbody>
  </table>
);

export default OrderTable;
```
**Refactored Code and Folder Structure**

I have refactored the code and updated the folder structure to improve organization and readability. The changes include:

1. Created a new "hooks" folder to store custom hooks, and placed the `useFetchOrders` hook inside it.

2. Moved the `OrderRow` and `OrderTable` components into separate files inside the "components/orders" folder.

3. Utilized the `useFetchOrders` hook in the `App` component to fetch and handle orders data.

The new folder structure now looks like this:

src/
|-- components/
| |-- orders/
| | |-- OrderRow.js
| | |-- OrderTable.js
| |-- App.js
|-- hooks/
| |-- useFetchOrders.js
|-- utils/
| |-- dateUtils.js
|-- index.js

These changes improve code modularity and maintainability by separating concerns into their respective folders. The custom hook `useFetchOrders` handles data fetching, and the `App` component uses this hook to manage the loading and error states for order data.

The code is now more organized, making it easier to understand and maintain.

### **Issue 2: "As a bakery owner, I should be able to place a sheet with multiple cookies into an oven"**

#### Changes Made

1. **Pluralization of Model Name:**
   - Added the following code to `config/initializers/inflections.rb` to handle the pluralization of the model name:
     ```ruby
     ActiveSupport::Inflector.inflections(:en) do |inflect|
       inflect.irregular 'cookie', 'cookies'
     end
     ```

2. **Model Modifications:**
   - Updated the `Oven` model to have a `has_many :cookies` association. This change allows an oven to hold multiple cookies simultaneously, enabling us to place multiple cookies into the oven together.

3. **Quantity Field in Cookie Form:**
   - Added a virtual attribute `quantity` field to the cookie form. Now, when preparing cookies for baking, the user can specify the quantity of cookies they want to bake with the same fillings.

4. **Cookie Baking:**
   - Created the `Cookies/CreateService` to support baking multiple cookies with the same fillings. The service can now create the specified number of cookies in the oven, all with the same fillings. Custom validation ensures the quantity is greater than zero.

5. **Oven Show Component:**
   - Created a view component called `Oven::ShowComponent` to display the oven's information, including the number of cookies in the oven and the fillings of the first cookie (if any). The component enhances the organization and reusability of the view logic.

6. **View Components for All Views and Form:**
   - Created view components for all oven-related views, including the index, show, and form views. The use of view components improves the maintainability and modularity of the view code.

7. **Testing:**
   - Implemented thorough testing for the `Cookies/CreateService` to ensure it correctly handles the creation of multiple cookies with the same fillings.
   - Wrote unit tests for the  view components to validate the information displayed in the views.
   - All tests are written using RSpec and follow best practices for testing Rails applications.
 
## Issue 3 and Issue 4: Enhancing Cookie Baking Process and Real-time Oven Updates

This pull request addresses two issues:
- Issue 3: Cookies are now actually cooked in the bakery application.
- Issue 4: As a bakery owner, the oven page now updates automatically when cookies are ready.

### Changes Made

1. Added a new virtual attribute for cooking time:
   - A virtual attribute "cooking_time" has been introduced in the form.
   - When creating cookies, the baking time is now set using the "cooking_time" attribute.

2. Implementing Stimulus Controller for Real-time Updates:
   - Utilized the Stimulus Controller to achieve dynamic updates.
   - Displayed a loader and a timer until the baking time elapses.
   - Implemented polling to periodically check the cookies' status.

3. Button to Retrieve Cookies:
   - After the baking time is completed, a "Retrieve Cookies" button appears.



4. Achieving Real-time Updates:
   - Leveraged Turbo Streams to achieve real-time updates.
   - The oven page now automatically reflects changes in cookie status.

### Technical Details

- Introduced a virtual attribute "cooking_time" to accurately set baking time.
- Implemented a Stimulus Controller to manage real-time updates:
   - Displayed a loader and countdown timer until the baking time elapses.
   - Utilized polling to monitor the cookies' status.
   - Displayed the "Retrieve Cookies" button once baking is completed.
- Integrated Turbo Streams to achieve real-time updates on the oven page.
These enhancements provide an improved user experience, ensuring that cookies are accurately baked and the oven page updates in real-time, enhancing efficiency and convenience for bakery owners.
##  Issue 5 - Add Loading Indicator for Order Listing and Issue 6 - Sorting Order Listing Table


### Issue 5: Add Loading Indicator for Order Listing

- Added a loading indicator that appears while waiting for data from the API.
- Improved user experience by providing visual feedback during data retrieval.

### Issue 6: Sorting Order Listing Table

- Implemented sorting functionality for the orders in ascending (asc) and descending (desc) order.
- Users can now easily organize and view orders based on their preferences.

### Code Refactoring and Component Testing

- Refactored code to improve readability and maintainability.
- Thoroughly tested the OrdersTable and OrdersPage components to ensure functionality and reliability.

These enhancements enhance the user experience by introducing a loading indicator for data retrieval and enabling efficient sorting of order listings. The codebase has been refactored for better organization, and comprehensive testing has been performed to ensure the stability and correctness of the OrdersTable and OrdersPage components.

## Pull Request: Issue 7 - Marking Orders Fulfilled

### Description

This pull request addresses Issue #7 by implementing the ability to mark orders as fulfilled using the `fulfilled` attribute in the orders database column. It introduces a new "Fulfill Order" action, allowing users to mark orders as fulfilled, and ensures that the changes are properly saved to the database.

### Changes Made

- Added a "Fulfill Order" action to the order management interface.
- Utilized the `fulfilled` attribute in the orders database column to manage order fulfillment status.
- Implemented necessary backend logic to update the fulfillment status in the database.
- Created corresponding frontend components and UI elements to support order fulfillment.
- Added unit tests to validate the correctness of the new functionality and ensure proper handling of order fulfillment actions.

### Additional Changes

In addition to the primary implementation, the following enhancements have been made:

- Improved UI/UX for order management to clearly indicate fulfillment status.
- Added loading indicators to provide visual feedback during the fulfillment process.
- Enhanced error handling for scenarios where order fulfillment encounters issues.
- Ensured that order details accurately display the current fulfillment status.

### Testing

Thorough testing has been conducted to validate the new functionality:

- Unit tests have been included to cover the implementation of the "Fulfill Order" action.
- Integration tests have been executed to verify seamless interaction between frontend and backend components.
- Edge cases and potential errors have been meticulously tested to ensure the system's robustness.

This pull request presents a substantial enhancement to the order management system, resulting in an improved user experience and precise tracking of order fulfillment status.

Your review and feedback are greatly appreciated.

Thank you.

### Update:

In this update:

- Replaced the script-based test execution in `package.json` with a Jest configuration file and a setup file in the root directory for more organized test management.

- Introduced a testing helper for testing Stimulus controllers, streamlining the testing process and ensuring consistent controller behavior.

- Added a new test for the `CookiesController` in Stimulus to verify its functionality and behavior.

- Restored the `Pagination` component to the codebase and conducted debugging to ensure proper integration with fulfillment actions when used in conjunction with paginations.

- Eliminated the usage of data-test-id attributes from the `OrderRaw` component, improving test readability and simplifying maintenance. The corresponding test for this component was also updated accordingly.

- Addressed React update warnings by wrapping asynchronous operations with the `act` function, ensuring that state updates are handled correctly during testing.

- Integrated Jest tests into GitHub Actions for automated testing and continuous integration.

### New Update:

- Refactored controllers and services for improved validation and real-time data display. Modified filling process to utilize icons and added Stimulus controllers for buttons and form reset. Updated tests. Known bug: Retrieving cookies after timer expiry requires two clicks - first for the "Retrieve" button and second for the "Oven is empty" message. Recognized antipattern: Using inline scripts for Tailwind CSS.

**Changes Made:**
- Controller and service code has been refactored to enhance validation and optimize real-time data rendering.
- Modified the way cookies are filled by incorporating icon usage, alongside the introduction of Stimulus controllers for buttons and form resetting.
- Test suite has been updated and improved to ensure robust functionality.
 -Added a GitHub Actions workflow to check and validate Tailwind CSS usage, ensuring adherence to best practices.

**Bug Notice:**
- A bug has been identified where two clicks are needed to retrieve cookies after the timer has expired and to display the "Oven is empty" message.

**Antipattern:**
- It has been recognized that using inline scripts for Tailwind CSS is not a recommended approach.

Thank you for your attention to these updates. We appreciate your ongoing collaboration and support.

Best regards,
Dejan

DeVuDeveloper committed:
works

DeVuDeveloper committed:
Added gems: simple_form, turbo-rails, stimulus-rails, redis, and sidekiq. Significant improvement in page rendering with turbo frames and turbo streams. Form validation implemented and code refactored with added view components. First tests completed before writing code.

DeVuDeveloper committed:
Each child in an array or iterator should have a unique key prop warning is resolved, and the React code is now better organized and structured

DeVuDeveloper committed:
created a functioning system where the bakery owner can inte…
…ract with the oven to bake multiple cookies with the same filling and then remove them from the oven when they are ready

DeVuDeveloper committed:
#Issue + Issue4: Added real-time cookie baking timer with hotwire and stimulus. Testing

DeVuDeveloper committed:
fixing linter errors

DeVuDeveloper committed:
Issue 5 + Issue 6: Added loader, sorting (ascending and descending), and tests for the OrdersPage and OrderTable components

DeVuDeveloper committed:
## Issue 7 - Marking Orders Fulfilled: adds functionality to mark orders as fulfilled using the  database column.

joelmoss commented on app/src/orders/components/OrderRow.jsx:
```diff
@@ -0,0 +1,53 @@
+
+  return (
+    <tr>
+      <td data-testid="order-id">{id}</td>
```
test data/attributes should not be in non-test code.

DeVuDeveloper committed:
resolved bug with pagination and added test for stimulus CookiesController

DeVuDeveloper committed:
git hub action for jset test add

DeVuDeveloper committed:
Update: Refactored OrderRow component and updated its test to remove data-test-id attributes

DeVuDeveloper commented:
Hi Joe,

Thank you for your prompt response and for reviewing the code. I had some
errors in the tests, which I added for debugging purposes. I've made
changes to both the component and the test, and I now have an update for
the pull request. I have some free time now, and I want to see how the Gem
Proscenium will collaborate with Tailwind CSS and Vite. After that, I'll
work on styling to make it resemble the Digital Oven design more closely.

Could you please take a look at the update? I'll also send you the update:
<https://github.com/DeVuDeveloper/test-app/pull/1>







*Replaced the script-based test execution in package.json with a Jest
configuration file and a setup file in the root directory for more
organized test management.Introduced a testing helper for testing Stimulus
controllers, streamlining the testing process and ensuring consistent
controller behavior.Added a new test for the CookiesController in Stimulus
to verify its functionality and behavior.Restored the Pagination component
to the codebase and conducted debugging to ensure proper integration with
fulfillment actions when used in conjunction with paginations.Eliminated
the usage of data-test-id attributes from the OrderRaw component, improving
test readability and simplifying maintenance. The corresponding test for
this component was also updated accordingly.Addressed React update warnings
by wrapping asynchronous operations with the act function, ensuring that
state updates are handled correctly during testing.Integrated Jest tests
into GitHub Actions.*

*Once again, thank you for your time.*

Best regards,
Dejan

On Tue, Aug 8, 2023 at 12:14 PM Joel Moss ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In app/src/orders/components/OrderRow.jsx
> <https://github.com/DeVuDeveloper/test-app/pull/1#discussion_r1286911138>:
>
> > +    setIsFulfilling(true);
> +
> +    try {
> +      await onFulfillOrder(order);
> +      setIsFulfilling(false);
> +      setCurrentFulfilledStatus(true);
> +      storeFulfilledStatus(id, true);
> +    } catch (error) {
> +      console.error('Error fulfilling order:', error);
> +      setIsFulfilling(false);
> +    }
> +  };
> +
> +  return (
> +    <tr>
> +      <td data-testid="order-id">{id}</td>
>
> test data/attributes should not be in non-test code.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DeVuDeveloper/test-app/pull/1#pullrequestreview-1566971450>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWLERYJ3BTJLCQCEKWDLD73XUIGR5ANCNFSM6AAAAAA26MHOOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

DeVuDeveloper committed:
refactoring using only ovens/index page for displaying everything in real time

DeVuDeveloper committed:
fix linter

DeVuDeveloper committed:
added stimulus test, changed validation is updating in index view in real time

DeVuDeveloper committed:
liter errors

DeVuDeveloper committed:
update readme

DeVuDeveloper committed:
added on/off status for oven and added priece for cookie. User hs to paid for amount choosed with cookies quantity.Working on back rest if cookies are overpaid

DeVuDeveloper committed:
Added balance functionality for users and enabled cookie creation if sufficient funds are available.

Additionally, implemented excess payment refund feature.

muhammad-ur-rehman commented on .eslintrc:
```diff
@@ -0,0 +1,21 @@
+        "tailwindcss/no-custom-classname": "warn",
+        "tailwindcss/no-contradicting-classname": "error"
+    }
+}
```
add empty line at the EOF

muhammad-ur-rehman commented on .github/workflows/linters.yml:
```diff
@@ -0,0 +1,105 @@
+        run: npm install
+
+      - name: Run Jest tests
+        run: npm test -- --ci --coverage
```
empty line EOF.
Please look at all of the files

muhammad-ur-rehman commented on .vscode/settings.json:
```diff
@@ -0,0 +1,2 @@
+{
+}
```
why we need this file?

muhammad-ur-rehman commented on README.md:
```diff
@@ -1,127 +1,42 @@
+<div style="display: flex; justify-content: space-between;">
+  <img src="app/assets/images/bakery1.png" alt="bakery 1" width="30%">
+  <img src="app/assets/images/bakery2.png" alt="bakery 2" width="30%">
+  <img src="app/assets/images/bakery3.png" alt="bakery 3" width="30%">
```
dont need to increase code size for the README file images are not required in readme until they are very important if it is important use any other free/paid source to place pictures there

muhammad-ur-rehman commented on __tests__/stimulus/cookies_controler.test.js:
```diff
@@ -0,0 +1,31 @@
+
+  expect(getHTML()).toMatchSnapshot();
+});
+
+
+
```
empty spaces

muhammad-ur-rehman commented on app/components/ovens/cookie_info_component.html.erb:
```diff
@@ -0,0 +1,21 @@
+    <% if @oven.cookies.any? { |cookie| cookie.cooked_at.present? } %>
+      <div>
+        <p class="text-orange-500">
+          <%= pluralize(@oven.cookies.size, "cookie") %> with <%= @oven.first_cookie_fillings || "no fillings" %>
```
have you handled `@oven.cookies` (n+1) in the code base?

muhammad-ur-rehman commented on app/controllers/application_controller.rb:
```diff
@@ -1,3 +1,13 @@
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:email, :password, :password_confirmation)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:password, :password_confirmation, :current_password, :balance)
+    end
```
make these private i think we dont need them available in all controllers

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
-    if @oven.cookie
-      redirect_to @oven, alert: 'A cookie is already in the oven!'
+  def create
+    if @oven.online
+      if cookies_already_in_oven?
+        oven_already_has_cookies
+      else
+        paid_amount = params[:cookie][:price].to_f
+        create_service = Cookies::CreateService.new(current_user, @oven, cookie_params)
+        validation_service = Cookies::ValidationService.new(cookie_params)
+
+        if validation_service.validate.nil? && create_service.call
+          if current_user.balance >= paid_amount
+            current_user.subtract_from_balance(paid_amount)
+
+            respond_to do |format|
+              format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+              format.turbo_stream.append("cookies", partial: "cookies/created_cookie")
+            end
+          else
+            respond_to do |format|
+              format.html { redirect_to ovens_path, alert: "Insufficient funds." }
+              format.turbo_stream { flash.now[:alert] = "Insufficient funds." }
+            end
+          end
+        else
+          @cookie = @oven.cookies.build(cookie_params)
+          flash.now[:alert] = validation_service.validate || create_service.error_message
+          respond_to do |format|
+            format.html { render :new, alert: flash.now[:alert] }
+            format.turbo_stream { flash.now[:alert] = flash.now[:alert] }
```
complexity of this method is very high reduce it

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
+            current_user.subtract_from_balance(paid_amount)
+
+            respond_to do |format|
+              format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+              format.turbo_stream.append("cookies", partial: "cookies/created_cookie")
```
too many render lines there should be only 1 render for html, 1 for tubo etc.

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
+              format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+              format.turbo_stream.append("cookies", partial: "cookies/created_cookie")
+            end
+          else
+            respond_to do |format|
+              format.html { redirect_to ovens_path, alert: "Insufficient funds." }
+              format.turbo_stream { flash.now[:alert] = "Insufficient funds." }
+            end
+          end
+        else
+          @cookie = @oven.cookies.build(cookie_params)
+          flash.now[:alert] = validation_service.validate || create_service.error_message
+          respond_to do |format|
+            format.html { render :new, alert: flash.now[:alert] }
```
this method is doing magic?
you have to break this into smaller method follow SRP, DRY

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
-    redirect_to oven_path(@oven)
+  def claim_change
+    paid_amount = params[:paid_amount].to_f
+    total_price = params[:total_price].to_f
```
always permit params before using

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
-    @oven = current_user.ovens.find_by!(id: params[:oven_id])
-    @cookie = @oven.create_cookie!(cookie_params)
-    redirect_to oven_path(@oven)
+  def claim_change
```
if this method is not linked with any route make it private

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
+    change_to_claim = paid_amount - total_price
+
+    if change_to_claim.positive?
+      current_user.add_to_balance(change_to_claim)
```
what if this fails it will still return success

muhammad-ur-rehman commented on app/controllers/cookies_controller.rb:
```diff
@@ -1,24 +1,84 @@
 
   private
 
+  def cookies_already_in_oven?
+    @oven.cookies.present?
+  end
+
+  def oven_already_has_cookies
+    respond_to do |format|
+      format.html { redirect_to ovens_path, alert: "Cookies are already in the oven!" }
+      format.turbo_stream { flash.now[:alert] = "Cookies are already in the oven!" }
+    end
+  end
+
+  def cookies_created_successfully
+    respond_to do |format|
+      format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+      format.turbo_stream { flash.now[:notice] = "Cookies were successfully created." }
+    end
+  end
```
you need to slim the controller make use of concern or hellper or service what ever make sense

muhammad-ur-rehman commented on app/controllers/ovens_controller.rb:
```diff
@@ -1,17 +1,57 @@
     @oven = current_user.ovens.find_by!(id: params[:id])
-    @oven.cookie.update!(storage: current_user) if @oven.cookie
-    redirect_to @oven, alert: 'Oven emptied!'
+  end
+
+  def empty_oven_cookies
+    @oven.cookies&.update(storage: current_user)
+  end
+
+  def respond_to_oven_empty
+    respond_to do |format|
+      format.html { redirect_to oven_path(@oven) }
+      format.turbo_stream
+    end
+  end
+
+  def oven_status_json
+    {ready: @oven.first_cookie_ready?, time_left: @oven.time_left_for_first_cookie}
   end
```
code reusability + silm all the controllers

muhammad-ur-rehman commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,20 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!, only: [:show, :update, :add_balance]
```
this controller only have 1 method and you are authenticating for 3 methods

muhammad-ur-rehman commented on app/javascript/controllers/application.js:
```diff
@@ -0,0 +1,8 @@
+application.debug = false
+window.Stimulus   = application
+
+export { application }
```
extra line EOF

muhammad-ur-rehman commented on app/javascript/controllers/cookies_controller.js:
```diff
@@ -0,0 +1,60 @@
+      this.fetchOvenStatus();
+    }, 5000);
+  }
+
+  
```
setup linter for these also

muhammad-ur-rehman commented on app/javascript/entrypoints/application.js:
```diff
@@ -0,0 +1,28 @@
+
+console.log('Visit the guide for more information: ', 'https://vite-ruby.netlify.app/guide/rails')
+
+// Example: Load Rails libraries in Vite.
+//
+// import * as Turbo from '@hotwired/turbo'
+// Turbo.start()
+//
+// import ActiveStorage from '@rails/activestorage'
+// ActiveStorage.start()
+//
+// // Import all channels.
+// const channels = import.meta.globEager('./**/*_channel.js')
+
+// Example: Import a stylesheet in app/frontend/index.css
+// import '~/index.css'
```
remove unnecessary comments

muhammad-ur-rehman commented on app/javascript/entrypoints/application.js:
```diff
@@ -0,0 +1,28 @@
+// views/layouts/application.html.erb
+//
+//    <%= vite_client_tag %>
+//    <%= vite_javascript_tag 'application' %>
+console.log('Vite ⚡️ Rails')
+
+// If using a TypeScript entrypoint file:
+//     <%= vite_typescript_tag 'application' %>
+//
+// If you want to use .jsx or .tsx, add the extension:
+//     <%= vite_javascript_tag 'application.jsx' %>
+
+console.log('Visit the guide for more information: ', 'https://vite-ruby.netlify.app/guide/rails')
+
```
you are printing this on console for production as well?

muhammad-ur-rehman commented on app/services/cookies/create_service.rb:
```diff
@@ -0,0 +1,31 @@
+  def initialize(user, oven, params)
+    @user = user
+    @oven = oven
+    @params = params
```
how you know what variables you need in params

muhammad-ur-rehman commented on app/services/cookies/create_service.rb:
```diff
@@ -0,0 +1,31 @@
+
+    num_cookies_to_bake.times do
+      cookie = @oven.cookies.build(@params)
+      cookie.save
+      cookie.update(cooked_at: Time.now + cooking_time)
```
what if this breaks?

muhammad-ur-rehman commented on app/src/orders/App.jsx:
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import OrdersPage from "./pages/OrdersPage";
+
+const App = () => {
+  return (
+    <div>
+      <OrdersPage />
+    </div>
+  );
+};
+
+export default App;
+
+
+
+
```
linting issues

muhammad-ur-rehman commented on app/src/orders/components/OrderRow.jsx:
```diff
@@ -0,0 +1,58 @@
+};
+
+export default OrderRow;
+
+
+
+
+
+
```
linting

muhammad-ur-rehman commented on app/src/orders/hooks/UseFetchOrders.jsx:
```diff
@@ -0,0 +1,31 @@
+};
+
+export default useFetchOrders;
+
+
+
+
+
+
```
linting

muhammad-ur-rehman commented on app/views/layouts/_navigation_links.html.erb:
```diff
@@ -6,7 +6,7 @@
   </li>
   <li class="nav-item">
-    <%= link_to 'Order listing', orders_path, class: 'nav-link' %>
+    <%= link_to 'Order listing', orders_path, class: 'nav-link', data: {turbo: false} %>
```
one question why you need the turbo fase?

DeVuDeveloper replied:
It is React, so I don't need turbo.

muhammad-ur-rehman commented on app/views/layouts/application.css:
```diff
@@ -1,5 +1,63 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 0.75rem;
+
+  max-width: 100%;
+  width: max-content;
+  padding: 0 1rem;
+}
+
+.flash__message {
+  font-size: 0.875rem;
+  animation: appear-then-fade 4s both;
+  border-radius: 999px;
+  margin-top: -10px;
+  margin-right: 100px;
+}
+
+.flash__message-notice {
+  color: hsl(120, 96%, 53%);
```
make use of variables in scss

muhammad-ur-rehman commented on app/views/ovens/_toggle.html.erb:
```diff
@@ -0,0 +1,20 @@
+          </svg>
+        </button>
+      <% end %>
+
+
+
+
```
linting issues in all of the HTML files

DeVuDeveloper committed:
update readme
